### PR TITLE
Add Job.is_available()

### PR DESCRIPTION
**Pull request by rhysd**
I added it because I want to check job feature is available in advance.
For example, it can be used to determine default configuration.

```vim
let g:plugin_execute_async = get(g:, 'plugin_execute_async', s:Job.is_available())
```

### CODE REVIEW

File: autoload/vital/__vital__/System/Job.vim [hunk 1]
```diff
@@ -81,6 +84,10 @@ else
     return job
   endfunction
 
+  function! s:is_available() abort
+    return has('job') && has('patch-8.0.0027')
```
Comment (lambdalisue) [inline autoload/vital/__vital__/System/Job.vim:88]: v8.0.0027 is required only for `Job.wait()` so please change the function name to more specific to `Job.wait()` (`is_wait_available()` or whatever)

File: doc/Vital/System/Job.txt [hunk 2]
```diff
@@ -82,6 +82,9 @@ start({args}[, {options}])
 	Note that a first argument of the callback is a builtin job instance
 	in Vim or a job id number in Neovim.
 
+					*Vital.System.Job.is_available()*
+is_available()
+	Return true if job feature is available. Otherwise return false.
```
Comment (lambdalisue) [inline doc/Vital/System/Job.txt:87]: And make the documentation more clear :+1:

**Review by lambdalisue — CHANGES_REQUESTED** [review]

**Comment by rhysd** [comment]
Ah, well, the behavior is not what I wanted. I wanted to know job feature is available or not. I knew that `has('job')` is not sufficient to check wether job is available or not . I musunderstood that  `has('patch-8.0.0027')` checks job feature's availability. I think I should check that on my own in my plugin. Thanks for your review.

**Comment by rhysd** [comment]
And I should likely use [this](https://github.com/lambdalisue/vital-System-Job/blob/master/autoload/vital/__vital__/System/Process/Job.vim#L11) :+1:

**Pull request closed without merging**